src/main/java/com/device/shop/controller/PaymentDetailsController.java

```diff
@@ -0,0 +1,44 @@
+package com.device.shop.controller;
+
+
+import com.device.shop.entity.PaymentDetails;
+import com.device.shop.service.PaymentDetailsService;
+
+import lombok.AllArgsConstructor;
+
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.*;
+
+@Controller
+@AllArgsConstructor
+public class PaymentDetailsController {
+
+    PaymentDetailsService paymentDetailsService;
+
+    @PostMapping("/{paymentId}")
+    public ResponseEntity<PaymentDetails> createPaymentById(@PathVariable("paymentId") Long id) {
+        PaymentDetails payment = paymentDetailsService.createPaymentById(id);
+        return ResponseEntity.ok(payment);
+    }
+
+    @GetMapping("/{paymentId}")
+    public ResponseEntity<PaymentDetails> getPaymentDetaById(@PathVariable("paymentId") Long paymentId) {
+        PaymentDetails payment = paymentDetailsService.getPaymentById(paymentId);
+        return ResponseEntity.ok(payment);
+    }
+
+    @PutMapping("/{id}")
+    public ResponseEntity<PaymentDetails> updateOrderDetails(@PathVariable("id") Long id, @RequestBody PaymentDetails paymentDetails) {
```

Comment:
Change the name of the method, this controller is definitely not about order details to use the name you've provided